clients/vscode/src/inline-edit/quickPick.ts

```diff
@@ -0,0 +1,321 @@
+    this.client.chat.provideEditCommands(
+      { location: this.editLocation },
+      { commands: this.suggestedCommand, callback: () => this.updateQuickPickList() },
+      this.fetchingSuggestedCommandCancellationTokenSource.token,
+    );
+  }
+
+  private async handleTriggerItemButton(event: QuickPickItemButtonEvent<CommandQuickPickItem>) {
+    const item = event.item;
+    const button = event.button;
+    if (button.iconPath instanceof ThemeIcon && button.iconPath.id === "settings-remove") {
+      const index = this.recentlyCommand.indexOf(item.value);
+      if (index !== -1) {
+        this.recentlyCommand.splice(index, 1);
+        await this.config.updateChatEditRecentlyCommand(this.recentlyCommand);
+        this.updateQuickPickList();
+      }
+    }
+
+    if (button.iconPath instanceof ThemeIcon && button.iconPath.id === "edit") {
+      this.updateQuickPickValue(item.value);
+    }
+  }
+}
+
+export class FileSelectionQuickPick {
+  quickPick = window.createQuickPick();
+  private maxSearchFileResult = 30;
+  private resultDeffer = new Deferred<string | undefined>();
+
+  constructor(private editor: TextEditor) {}
+
+  private get workspaceFolder() {
+    return workspace.getWorkspaceFolder(this.editor.document.uri);
+  }
+
+  start() {
+    this.quickPick.title = "Select File";
+    this.quickPick.placeholder = "Enter file name to search";
+    this.quickPick.buttons = [QuickInputButtons.Back];
+    this.quickPick.ignoreFocusOut = true;
+    this.quickPick.onDidChangeValue((e) => this.updateFileList(e));
+    this.quickPick.onDidAccept(() => this.handleAccept());
+    this.quickPick.onDidHide(() => this.handleHidden());
+    this.quickPick.onDidTriggerButton((e) => this.handleTriggerButton(e));
+    this.quickPick.show();
+    this.updateFileList("");
+    return this.resultDeffer.promise;
+  }
+
+  private async updateFileList(val: string) {
+    this.quickPick.busy = true;
+    const fileList = await this.searchFilelist(val);
+    this.quickPick.items = fileList;
+    this.quickPick.busy = false;
+  }
+
+  private handleAccept() {
+    const file = this.quickPick.selectedItems[0]?.label;
+    this.resultDeffer.resolve(file);
+  }
+
+  private handleHidden() {
+    this.resultDeffer.resolve(undefined);
+  }
+
+  private handleTriggerButton(e: QuickInputButton) {
+    if (e === QuickInputButtons.Back) {
+      this.quickPick.hide();
+    }
+  }
+
+  private async searchFileList(searchQuery: string) {
+    if (!this.workspaceFolder) {
+      return [];
+    }
+    const globPattern = caseInsensitivePattern(searchQuery);
+    const fileList = await this.findFiles(globPattern, { maxResults: this.maxSearchFileResult });
+    return fileList.map((file) => workspace.asRelativePath(file));
+  }
+
+  private findFiles = wrapCancelableFunction(
+    findFiles,
+    (args) => args[1]?.token,
+    (args, token) => [args[0], { ...args[1], token }] as Parameters<typeof findFiles>,
+  );
+
+  private async searchFilelist(query: string) {
```

Comment:
The different between funciton `searchFilelist` and `searchFileList` is only a `L`. So give it another name.
```suggestion
  private async getFileList(query: string) {
```